public/pages/Snapshots/components/RestoreSnapshotFlyout/RestoreSnapshotFlyout.tsx

```diff
@@ -0,0 +1,326 @@
+import AddPrefixInput from "../AddPrefixInput";
+import RenameInput from "../RenameInput";
+import SnapshotIndicesInput from "../SnapshotIndicesInput";
+import { ERROR_PROMPT } from "../../../CreateSnapshotPolicy/constants";
+
+interface RestoreSnapshotProps {
+  snapshotManagementService: SnapshotManagementService;
+  indexService: IndexService;
+  onCloseFlyout: () => void;
+  restoreSnapshot: (snapshotId: string, repository: string, options: object) => void;
+  snapshotId: string;
+}
+
+interface RestoreSnapshotState {
+  indexOptions: EuiComboBoxOptionOption<IndexItem>[];
+  selectedIndexOptions: EuiComboBoxOptionOption<IndexItem>[];
+  renameIndices: string;
+  prefix: string;
+  renamePattern: string;
+  renameReplacement: string;
+  listIndices: boolean;
+
+  repositories: CatRepository[];
+  selectedRepoValue: string;
+
+  snapshot: GetSnapshot | null;
+  snapshotId: string;
+  restoreSpecific: boolean;
+  restoreAliases: boolean;
+  partial: boolean;
+
+  repoError: string;
+  snapshotIdError: string;
+}
+
+export default class RestoreSnapshotFlyout extends Component<RestoreSnapshotProps, RestoreSnapshotState> {
+  static contextType = CoreServicesContext;
+  constructor(props: RestoreSnapshotProps) {
+    super(props);
+    this.state = {
+      indexOptions: [],
+      selectedIndexOptions: [],
+      renameIndices: "add_prefix",
+      prefix: "",
+      renamePattern: "",
+      renameReplacement: "",
+      listIndices: false,
+      repositories: [],
+      selectedRepoValue: "",
+      snapshot: null,
+      snapshotId: "",
+      restoreSpecific: false,
+      restoreAliases: true,
+      partial: false,
+      repoError: "",
+      snapshotIdError: "",
+    };
+  }
+
+  async componentDidMount() {
+    await this.getRepos();
+    await this.getIndexOptions();
+  }
+
+  onClickAction = () => {
+    const { restoreSnapshot, snapshotId } = this.props;
+    const {
+      selectedRepoValue,
+      restoreSpecific,
+      selectedIndexOptions,
+      indexOptions,
+      snapshot,
+      renameIndices,
+      prefix,
+      renamePattern,
+      renameReplacement,
+    } = this.state;
+    const selectedIndices = selectedIndexOptions.map((option) => option.label).join(",");
+    const allIndices = indexOptions.map((option) => option.label).join(",");
+    const pattern = renameIndices === "add_prefix" ? "(?<![^ ])(?=[^ ])" : renamePattern;
+
+    const options = {
+      indices: restoreSpecific ? selectedIndices : allIndices,
+      ignore_unavailable: snapshot?.ignore_unavailable || false,
+      include_global_state: snapshot?.include_global_state,
+      rename_pattern: pattern,
+      rename_replacement: renameIndices === "add_prefix" ? prefix : renameReplacement,
+      include_aliases: snapshot?.restore_aliases || false,
+      partial: snapshot?.partial || false,
+    };
+    let repoError = "";
+
+    if (!snapshotId.trim()) {
+      this.setState({ snapshotIdError: "Required" });
```

Comment:
Minor detail, but I believe the convention is for `helpText` and error messages to end with a hard stop (`.`).

Reply:
fixed and pushed